### PR TITLE
feat(WebGL2): add order independent translucent rendering

### PR DESCRIPTION
**Pull request by martinken**
Add a basic order independent translucent rendering technique
for the WebGL backend. Requires WebGL2 support and a common
extension to work. No changes required in existing code as
this is now the default for webgl2.

Addresses https://github.com/Kitware/vtk-js/issues/1291

### CODE REVIEW

File: Sources/Rendering/OpenGL/Framebuffer/index.js [hunk 1]
```diff
@@ -242,7 +247,8 @@ export function extend(publicAPI, model, initialValues = {}) {
   // Build VTK API
   macro.obj(publicAPI, model);
 
-  macro.setGet(publicAPI, model, ['colorTexture']);
+  model.colorTextures = [];
```
Comment (finetjul) [inline Sources/Rendering/OpenGL/Framebuffer/index.js:250]: by doing so, you prevent passing color textures in constructor.
```
function defaultValues (initialValues) {
  return {
  // openGLRenderWindow: null,
  // glFramebuffer: null,
  colorTextures: [],
  // depthTexture: null,
  previousDrawBinding: 0,
  previousReadBinding: 0,
  prev...,
  ..initialValues
  };
}
function extend(...) {
  Object.assign(model, defaultValues(initialValues));
```

**Comment by finetjul** [comment]
Will it work in case of 2 translucent actors ? (i.e. replaces depth peeling ?)

File: Sources/Rendering/OpenGL/Framebuffer/index.js [hunk 2]
```diff
@@ -242,7 +247,8 @@ export function extend(publicAPI, model, initialValues = {}) {
...
+  macro.setGet(publicAPI, model, ['colorTextures']);
```
[the remark this answers is not in the document]
Comment (martinken) [reply Sources/Rendering/OpenGL/Framebuffer/index.js:251]: Yes, should be just getArray. Fixed.

**Comment by martinken** [comment]
Yes, handles multiple props.

File: Sources/Rendering/OpenGL/Framebuffer/index.js (hunk 1 above)
Comment (martinken) [reply Sources/Rendering/OpenGL/Framebuffer/index.js]: Updated to error on this case and cleaned up the naming and added some error checks.

**Comment by finetjul** [comment]
LGTM

**Comment by martinken** [comment]
@berkgeveci  FYI here is the translucent support for WebGL2

**Comment by finetjul** [comment]
@martinken can you please share some doc/reference on the technique being used here. I have a mismatch between the result rendering and what I expect when cells are opaque (alpha=1).

**Comment by martinken** [comment]
You mean a transparent object with some opaque cells I'm guessing. That will not look correct. The algorithm can be thought of as an opacity and depth weighted average of colors for all translucent pixels. The resulting total opacity is correct but the resulting color is a heuristic. This post gives the basics http://casual-effects.blogspot.com/2014/03/weighted-blended-order-independent.html though our weighting by depth may be different.

**Pull request merged**